### DISCUSSION TITLE
Client credentials enabled?

### DISCUSSION THREAD

**Discussion by afterthought** [zitadel/zitadel#2535] [Q&A] [answered]
Hi. The docs says that this is supported. However, the openid-configuration endpoint does not list this as a supported grant_type. I tested and confirmed that the api endpoint returns a 400, not supported error. I was just trying to use this method during the early stages of a project for demos. Is [this documentation](https://docs.zitadel.ch/docs/apis/openidoauth/grant-types) wrong?

**Comment by fforootd** [comment, marked as answer]
Hi @afterthought  

We support `client credentials` but only for `clients` and not  `service accounts`.
@livio-a @fgerschwiler please correct me if I had this one wrong.

Could you provide us some more details on what did not work?
e.g. Framework, Language, OIDC Settings, ...

**Reply by afterthought** [reply]
Well, the plan was to use the `client`. I was testing from swaggerUI, but postman would probably work too. I had set it up with the "Basic" authentication mode.

**Reply by fforootd** [reply]
cc @livio-a @fgerschwiler

**Reply by livio-a** [reply]
Hey @afterthought 

There is definitely a mismatch between the documentation and the implementation. At the moment client_credentials grant is not supported.
But I will look at it tomorrow with my colleague and give you feedback.

**Reply by livio-a** [reply]
Hey @afterthought 

I checked and we definitely don't support client_credentials. I will fix the documentation (https://github.com/caos/zitadel/pull/2586).
But since you're using a swaggerUI / postman I'd recommend using authorization code flow, which i believe you have done.
We there provide authentication using basic auth or form post.

Using code flow you can also sign in with different users and grant them different roles in your project or in ZITADEL itself.

**Reply by livio-a** [reply]
Hey @afterthought 

We've just released [version 2.19](https://github.com/zitadel/zitadel/releases/tag/v2.19.0) with support for client_credentials on service users. If you run you own ZITADEL instance, you can already try it out: https://zitadel.com/docs/guides/integrate/client-credentials
We will deploy it to zitadel.cloud on Monday evening.
